server/command.go

```diff
@@ -317,6 +327,58 @@ func (p *Plugin) executeShowCommand(args *model.CommandArgs) (*model.CommandResp
 	return &model.CommandResponse{}, nil
 }
 
+func (p *Plugin) executeSyncCommand(args *model.CommandArgs, parameters []string) (*model.CommandResponse, *model.AppError) {
+	if !p.API.HasPermissionTo(args.UserId, model.PermissionManageSharedChannels) {
+		return p.cmdError(args.UserId, args.ChannelId, "Unable to execute the command, you require `manage_shared_channels` permission to sync the channel.")
+	}
+
+	channel, appErr := p.API.GetChannel(args.ChannelId)
+	if appErr != nil {
+		return p.cmdError(args.UserId, args.ChannelId, "Unable to get the current channel information.")
+	}
+
+	if channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
+		return p.cmdError(args.UserId, args.ChannelId, "Syncrhonizing DMs or GMs is not supported yet")
+	}
+
+	hours := 24
+	if len(parameters) == 1 {
+		var err error
+		hours, err = strconv.Atoi(parameters[0])
+		if err != nil {
+			return p.cmdError(args.UserId, args.ChannelId, "Invalid command, please pass a valid number of hours as a parameter.")
+		}
+	} else if len(parameters) > 1 {
+		return p.cmdError(args.UserId, args.ChannelId, "Invalid command, please pass the number of hours as a parameter.")
+	}
+
+	link, err := p.store.GetLinkByChannelID(args.ChannelId)
+	if err != nil || link == nil {
+		return p.cmdError(args.UserId, args.ChannelId, "Link doesn't exist.")
+	}
+
+	if _, err := p.msteamsAppClient.GetTeam(link.MSTeamsTeam); err != nil {
+		return p.cmdError(args.UserId, args.ChannelId, "Unable to get the MS Teams team information.")
+	}
+
+	if _, err := p.msteamsAppClient.GetChannelInTeam(link.MSTeamsTeam, link.MSTeamsChannel); err != nil {
+		return p.cmdError(args.UserId, args.ChannelId, "Unable to get the MS Teams channel information.")
+	}
+
+	p.sendBotEphemeralPost(args.UserId, args.ChannelId, fmt.Sprintf("Synchronizing last %d hours of the channel...", hours))
+	go func() {
+		since := time.Now().Add(-time.Duration(hours) * time.Hour)
+		p.API.LogDebug("Running syncrhonization for channel", "teamID", link.MSTeamsTeam, "channelID", link.MSTeamsChannel, "since", since)
```

Comment:
typo: `syncrhonization` should be `synchronization`, also propose we log the `args.ChannelId` to connect it on the Mattermost side.